### PR TITLE
Add ResourceTransactionTestCase and mixin.

### LINKED ISSUE #684: Turn ResourceTestCase into mixin (to use with TransactionTestCase)
tastypie provides handy `tastypie.test.ResouceTestCase` class.

However it is inherited from `TestCase` so you cannot use it with `TransactionTestCase`.

Please make it a mixin so you can use these handy methods with any TestCase class you want.

Alternatively, create a separate class `ResouceTransactionTestCase`.

### PR DESCRIPTION
**Pull request by jcushman**
Fixes #684

### CODE REVIEW

File: tastypie/test.py [hunk 1]
```diff
@@ -572,3 +573,18 @@ def assertKeys(self, data, expected):
         changes.
         """
         self.assertEqual(sorted(data.keys()), sorted(expected))
+
+
+
+class ResourceTestCase(ResourceTestCaseMixin, TestCase):
```
Comment (dotsbb) [inline tastypie/test.py]: `579:1: E303 too many blank lines` [#1](https://travis-ci.org/django-tastypie/django-tastypie/jobs/97561980)

File: tastypie/test.py [hunk 2]
```diff
@@ -572,3 +573,18 @@ def assertKeys(self, data, expected):
...
+    Subclass Django's TestCase class with useful methods for testing Tastypie APIs.
+    """
+    pass
+
+
+class ResourceTransactionTestCase(ResourceTestCaseMixin, TransactionTestCase):
+    """
+    Subclass Django's TransactionTestCase class with useful methods for testing Tastypie APIs.
+    """
+    pass
```
Comment (dotsbb) [inline tastypie/test.py]: `./tastypie/test.py:590:9: W292 no newline at end of file` [#2](https://travis-ci.org/django-tastypie/django-tastypie/jobs/97561980)

**Comment by SeanHayes** [comment]
Instead of adding ResourceTransactionTestCase, I'd prefer to deprecate ResourceTestCase by adding a warning in it's init or setUp method..

**Comment by jcushman** [comment]
How's this?

**Comment by SeanHayes** [comment]
Awesome, thanks!

**Pull request merged**